config/v1/types_authentication.go

```diff
@@ -43,13 +43,24 @@ type AuthenticationSpec struct {
 	// +optional
 	OAuthMetadata ConfigMapNameReference `json:"oauthMetadata"`
 
-	// webhookTokenAuthenticators configures remote token reviewers.
-	// These remote authentication webhooks can be used to verify bearer tokens
+	// webhookTokenAuthenticators is DEPRECATED, setting it has no effect.
+	WebhookTokenAuthenticators []WebhookTokenAuthenticator `json:"webhookTokenAuthenticators,omitempty"`
+
+	// webhookTokenAuthenticator configures a remote token reviewer.
+	// This remote authentication webhook can be used to verify bearer tokens
 	// via the tokenreviews.authentication.k8s.io REST API.  This is required to
 	// honor bearer tokens that are provisioned by an external authentication service.
-	// The namespace for these secrets is openshift-config.
+	//
+	// The namespace for the referenced secret is openshift-config. It should
+	// contain kube config file data which describes how to access the remote
+	// webhook service. For further details, see:
+	//     https://kubernetes.io/docs/reference/access-authn-authz/authentication/#webhook-token-authentication
+	//
+	// The key "kubeConfig" is used to locate the data within the referenced secret.
+	// If the secret or expected key is not found, the webhook is not honored.
+	// If the specified kube config data is not valid, the webhook is not honored.
 	// +optional
-	WebhookTokenAuthenticators []WebhookTokenAuthenticator `json:"webhookTokenAuthenticators,omitempty"`
+	WebhookTokenAuthenticator SecretNameReference `json:"webhookTokenAuthenticator"`
```

Comment:
webhookTokenAuthenticatorSecret ?

Reply:
or even webhookTokenAuthenticatorKubeconfigSecret.

Why are we sure that we will never need any other fields for the webhook. I would feel better with a struct with a secret name as first field, but extensible later.

Reply:
I don't think we specify the resource type (CM/Secret) anywhere else in the name of the field. But `WebhookTokenAuthenticatorKubeconfig` might actually work better since the option we are mapping this field to is `authentication-token-webhook-config-file`. I'll improve the description to mention a secret to reference.

Reply:
Or perhaps keep the old struct for the extensibility, that's perhaps even a better idea

Reply:
+1